compiler/sem.nim

```diff
@@ -375,6 +375,11 @@ proc semConstExpr(c: PContext, n: PNode; expectedType: PType = nil): PNode =
     return n
   if e.kind in nkSymChoices and e[0].typ.skipTypes(abstractInst).kind == tyEnum:
     return e
+  if e.typ.kind == tyProc and e.typ.callConv == ccClosure and {tfHasAsgn} * e.typ.flags != {}:
+    localError(c.config, n.info, "const closure proc with an environment cannot be used at runtime: " &
+                   renderTree(n, {renderNoComments}))
+    return e
+
```

Comment:
Instead you need to inspect the **value**! `e[2].kind` must be `nkNilLit` or similar.

Reply:
I dont understand, neither `e` or `n` length greater than 2 in the test case.

Reply:
Sorry, I meant `e[1].kind`

Reply:
```
nkIntLit
cast[CSighandlerT](0)
nkEmpty
proc () {.closure.} = echo ["Hello"]
nkSym
makeMyClosure(myRef)
```
it's I get from test case.